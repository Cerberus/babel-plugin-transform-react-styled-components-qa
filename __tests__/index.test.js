import * as babel from 'babel-core'
import path from 'path'

import plugin from '../src/index'

it('works', () => {
  const { code } = babel.transformFileSync(path.join(__dirname, 'code'), {
    plugins: [plugin]
  })
  expect(code).toMatchSnapshot()
})
