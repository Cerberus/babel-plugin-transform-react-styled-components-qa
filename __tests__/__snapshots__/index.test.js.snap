// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"\\"use strict\\";

/* eslint-disable */

// without attrs at ( standard html tags )
const Component = styled.div.attrs({
  \\"data-qa\\": \\"Component\\"
})\`
  font-size: 12px;
  border-bottom: \${props => props.underlined ? \\"1px solid\\" : none};
\`;

// with attrs ( standard html tags )
const Component2 = styled.div.attrs({
  underlined: true,
  accentColor: \\"red\\",
  \\"data-qa\\": \\"Component2\\"
})\`
  font-size: 12px;
  border-bottom: \${props => props.underlined ? \\"1px solid\\" : none};
\`;

// without attrs ( computed tag )
const Component5 = styled[tag].attrs({
  \\"data-qa\\": \\"Component5\\"
})\`
  width: 12px;
  border-bottom: \${props => props.underlined ? \\"1px solid\\" : none};
\`;

// with attrs ( computed tag )
const Component6 = styled[tag].attrs({
  underlined: true,
  accentColor: \\"blue\\",
  \\"data-qa\\": \\"Component6\\"
})\`
  width: 12px;
  border-bottom: \${props => props.underlined ? \\"1px solid\\" : none};
\`;"
`;
